pkg/cert/csr.go

```diff
@@ -59,3 +60,38 @@ func VendorCertificateRequest(vendorID core.PartyID, vendorName string, qualifie
 		ExtraExtensions: extensions,
 	}, nil
 }
+
+// CSRFromVendorCA generates a CSR based upon the VendorCA. It copies any extensions needed as well as the O and C
+// The common name is appended with the qualifier.
+func CSRFromVendorCA(ca *x509.Certificate, qualifier string, publicKey interface{}) (*x509.CertificateRequest, error) {
+	if ca == nil {
+		return nil, errors.New("missing vendor CA")
+	}
+
+	if strings.TrimSpace(qualifier) == "" {
+		return nil, errors.New("missing qualifier")
+	}
+
+	if publicKey == nil {
+		return nil, errors.New("missing public key")
+	}
+
+	// extract SAN and Domain extensions
+	extensions := []pkix.Extension{}
+
+	for _, e := range ca.Extensions {
+		if e.Id.Equal(OIDNutsDomain) || e.Id.Equal(OIDSubjectAltName) {
+			extensions = append(extensions, e)
+		}
+	}
+
+	return &x509.CertificateRequest{
+		Subject: pkix.Name{
```

Comment:
Why not just copy the whole subject and apply the replace on "CA"? Less error-prone: then it keeps working when the Vendor CA cert subject changes.

Also replacing all occurences of "CA" is also a bit tricky (what if my vendor is called "CARE Software B.V.", "CAREN", etc). Alternatives (which have drawbacks, too):

- You could say `Common Name = Organization + Qualifier` but that's also tricky in case `Common Name != Organization` (currently not the case though).
- You could check for postfix " CA" (including the space) and if it's there, remove it and append qualifier
- Let the caller provide it in the function call (caller can use registry to find the proper name): a bit cumbersome

Reply:
Used a replacement strategy